kod.go

```diff
@@ -0,0 +1,62 @@
+package ntp
+
+import "errors"
+
+/*
+
+This file contains KOD (Kiss-o-Death) Error type.
+See https://tools.ietf.org/html/rfc5905#section-7.4 for more detail
+
+*/
+
+var (
+	// ErrorACST the association belongs to a unicast server
+	ErrorACST = errors.New("the association belongs to a unicast server")
+	// ErrorAUTH server authentication failed
+	ErrorAUTH = errors.New("server authentication failed")
+	// ErrorAUTO autokey sequence failed
+	ErrorAUTO = errors.New("autokey sequence failed")
+	// ErrorBCST the association belongs to a broadcast server
+	ErrorBCST = errors.New("the association belongs to a broadcast server")
+	// ErrorCRYP cryptographic authentication or identification failed
+	ErrorCRYP = errors.New("cryptographic authentication or identification failed")
+	// ErrorDENY access denied by remote server
+	ErrorDENY = errors.New("access denied by remote server")
+	// ErrorDROP lost peer in symmetric mode
+	ErrorDROP = errors.New("lost peer in symmetric mode")
+	// ErrorRSTR access denied due to local policy
+	ErrorRSTR = errors.New("access denied due to local policy")
+	// ErrorINIT the association has not yet synchronized for the first time
+	ErrorINIT = errors.New("the association has not yet synchronized for the first time")
+	// ErrorMCST the association belongs to a dynamically discovered server
+	ErrorMCST = errors.New("the association belongs to a dynamically discovered server")
+	// ErrorNKEY no key found
+	ErrorNKEY = errors.New("no key found")
+	// ErrorRATE rate exceeded
+	ErrorRATE = errors.New("rate exceeded")
+	// ErrorRMOT alteration of association from a remote host running ntpdc
+	ErrorRMOT = errors.New("alteration of association from a remote host running ntpdc")
+	// ErrorSTEP a step change in system time has occurred, but the association has not yet resynchronized
+	ErrorSTEP = errors.New("a step change in system time has occurred, but the association has not yet resynchronized")
+
+	kodErrorMap = map[uint32]error{
+		stringToUint32("ACST"): ErrorACST,
```

Comment:
Suggestion: Consolidate the `ErrorFOO` var declarations with the error map definition.  For example:

```go
var kissErrors = map[uint32]error{
    stringToUint32("ACST"): new errors.New("association belongs to a unicast server"),
    // etc...
}
```
This will shorten the code significantly.  Also, I like the full error descriptions you have in this commit better than the shortened "KOD ACST" errors you added in the later commit.

Reply:
The error can't be exported like that.

Reply:
Hm. If we're going to export KOD errors, I'd want to export all of them. It wouldn't make sense to export just some.

I had another idea.  What if we added a `KissCode` field to the `Response` structure, like this:

```go
	// MinError is a lower bound on the error between the client and server
	// clocks. When the client and server are not synchronized to the same
	// clock, the reported timestamps may appear to violate the principle of
	// causality. In other words, the NTP server's response may indicate
	// that a message was received before it was sent. In such cases, the
	// minimum error may be useful.
	MinError time.Duration

	// KissCode is a 4-character string indicating the reason the server
	// issued a "kiss of death" (Stratum=0).
	KissCode string

	// Poll is the maximum interval between successive NTP polling messages.
	// It is not relevant for simple NTP clients like this one.
	Poll time.Duration
```

This way you can find out what the Kiss code is.  You can also use the string as a key for the kissErrors `map`.

Reply:
I think it would be more Go style if it's exported Error (i.e. io.EOF)

Reply:
I can still see a reason to support adding the `KissCode` field to the `Response`. If the server responds with an unrecognized kiss code, the user could then examine it.

Also, I am not quite ready to export all the possible error codes that the `ntp` module generates at this moment.  That's a big change that I'm not sure I agree is a useful thing to export. I also don't want to export only a subset of the errors (the kiss errors). 

Can we add the `KissCode` field for now and consider the exported errors as a possible future change?

Reply:
Since I never heard a response from you, I went ahead and introduced the `KissCode` response in PR https://github.com/beevik/ntp/pull/25. Let me know what you think.